src/components/SingleItem/SingleItem.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import styles from "./SingleItem.module.css";
+
+import { connect } from "react-redux";
+import { addToCart } from "../../redux/Shopping/shopping-actions";
+
+import Navbar from "../Navbar/Navbar";
+
+const SingleItem = ({ currentItem, addToCart }) => {
+  return (
+    <>
+      <Navbar />
+      <div className={styles.singleItem}>
+        <img
+          className={styles.singleItem__image}
+          src={currentItem.image}
+          alt={currentItem.title}
+        />
+        <div className={styles.singleItem__details}>
+          <p className={styles.details__title}>{currentItem.title}</p>
+          <p className={styles.details__description}>
+            {currentItem.description}
+          </p>
+          <p className={styles.details__price}>$ {currentItem.price}</p>
+
+          <button
+            onClick={() => addToCart(currentItem.id)}
+            className={styles.details__addBtn}
+          >
+            Add To Cart
+          </button>
+        </div>
+      </div>
+    </>
+  );
+};
+
+const mapStateToProps = state => {
```

Comment:
if you don't need dispatch to be passed into props, use short form straight into `connect((state) => ({ currItem: state... }))` wrap